python/ray/workflow/step_executor.py

```diff
@@ -195,16 +197,41 @@ async def _write_step_inputs(wf_storage: workflow_storage.WorkflowStorage,
 
 
 def commit_step(store: workflow_storage.WorkflowStorage, step_id: "StepID",
-                ret: Union["Workflow", Any], *,
-                exception: Optional[Exception]):
+                ret: Union["Workflow", Any], *, exception: Optional[Exception],
+                checkpoint: "CheckpointModeType"):
     """Checkpoint the step output.
     Args:
         store: The storage the current workflow is using.
         step_id: The ID of the step.
         ret: The returned object of the workflow step.
         exception: The exception caught by the step.
+        checkpoint: The checkpoint mode.
     """
     from ray.workflow.common import Workflow
+
+    if checkpoint == CheckpointMode.DEFAULT.value:
+        # Default value implies "SYNC".
```

Comment:
do we really need default here? It looks like the same as SYNC, so probably we can just use SYNC everywhere it is not specified.